python-lib/text_extraction/__init__.py

```diff
@@ -54,4 +54,108 @@ def extract_text_content(file_bytes, extension, with_pandoc):
             return file_bytes.decode()
         
         return text
-            
\ No newline at end of file
+
+
+def extract_text_chunks(filename, file_bytes, extension, with_pandoc):
+    if extension == "pdf":
+        pdf_pages = pdfium.PdfDocument(file_bytes)
+        return [
+            {'file': filename, 'text': page.get_textpage().get_text_range(), 'id': page_id + 1, 'metadata': page_id + 1, 'error_message': ""}
```

Comment:
See PR comment, I think we should return metadata in a more structured JSON object